VideoWeb/VideoWeb/ClientApp/src/app/services/conference/distributed-video-control-cache.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+import {
+    ApiClient,
+    ConferenceVideoControlStatuses,
+    SetConferenceVideoControlStatusesRequest,
+    SetConferenceVideoControlStatusesRequest_VideoControlStatusRequest
+} from '../clients/api-client';
+import { Logger } from '../logging/logger-base';
+import {
+    IHearingControlsState,
+    IParticipantControlsState,
+    IVideoControlCacheStorageService
+} from './video-control-cache-storage.service.interface';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DistributedVideoControlCacheService implements IVideoControlCacheStorageService {
+    private readonly loggerPrefix = '[DistributedVideoControlCacheService] -';
+
+    constructor(private apiClient: ApiClient, private logger: Logger) {}
+
+    private mapToRequest(hearingControlStates: IHearingControlsState) {
+        if (!hearingControlStates) {
+            return null;
+        }
+
+        const setConferenceVideoControlStatusesRequest = {
+            participant_id_to_video_control_status_map: {}
+        } as SetConferenceVideoControlStatusesRequest;
+
+        if (hearingControlStates) {
+            for (const key in hearingControlStates.participantStates) {
```

Comment:
Is key here participant id? If so might be good to give it a more descriptive name.

Reply:
@FarronF its a participant Id